### PR TITLE
[Ready] Re-adds M-90gl to Nuke Ops uplink

### PR DESCRIPTION
**Pull request by lilacrhododendron**
:cl: 
add: The M-90gl is back in the Nuke Ops uplink with rebalanced pricing. 
/:cl:

I thought this weapon was interesting enough to re-add as a Nuke Ops weapon. In the course of making this PR I went back to some of the PRs from when it was removed from the uplink to get a good sense of what was seen as wrong with it at the time. I have done my best to deal with these issues and balance it properly.
In terms of power, the weapon basically fits between the C-20r and the L6 SAW. It's more portable than a SAW, but generally less powerful, and costs just as much to purchase and outfit. The weapon fires half as fast as a SAW and the bullets do less damage, just barely critting an unarmored person with a full burst. Targets with even the relatively weak Security Officer body armor can withstand a burst from it, and people wearing Bulletproof Armor can take multiple bursts before going down. The grenade launcher is strong, but slow to reload, and expensive to use a lot.

For people who remember the old one, this is basically the same thing but significantly more expensive. I also changed the grenade shells to be purchasable individually for 2 TC each instead of buying a box of 4 for 5 TC. There's also no ridiculously cheap bundle this time around, that can stay gone.

### CODE REVIEW

**Comment by subject217** [comment]
I was thinking of reverting my mag cost change, back to 4 TC. Everything else is fine as I've priced it IMO.

**Comment by Borisvanmemes** [comment]
This is better then a L6-SAW imo, mostly due to the fact that it HAS GRANADES, AND THOSE STUN

**Comment by subject217** [comment]
>Better than the SAW

Not really. For one, it fires about half as fast, and doesn't do as much damage, which means that even people wearing basic body armor can survive a burst, as I mentioned. The lack of AP ammo makes it even worse, as people wearing bulletproof armor (or armor like the Captain's) will be pretty hard to bring down.

The stun is the only thing I really found concerning, and it's not that bad if you have teammates around you, and the AOE is pretty small.

**Comment by subject217** [comment]
>whereas the cr20 excels in one on one by being a burst-fire disabler 

The C-20r has not had stamina damage for two years. In fact, it has not had stamina damage since the very PR that removed the M-90.

**Comment by Supermichael777** [comment]
>when you show how much you role fluke ops

**Comment by MrDoomBringer** [comment]
for anyone curious as to what it looks like!
m90gl

**Comment by CosmicScientist** [comment]
Time to fulfil my stargate nukeop dreams.

File: code/modules/uplink/uplink_items.dm [hunk 1]
```diff
@@ -201,6 +201,15 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	cost = 10
 	surplus = 40
 	include_modes = list(/datum/game_mode/nuclear)
+	
+/datum/uplink_item/dangerous/carbine
+	name = "M-90gl Carbine"
+	desc = "A fully-loaded, specialized three-round burst carbine that fires 30-round 5.56mm magazines with a togglable \
```
Comment (ShizCalev) [inline code/modules/uplink/uplink_items.dm]: it fires magazines? now that's badass

File: code/modules/uplink/uplink_items.dm [hunk 2]
```diff
@@ -508,6 +517,22 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	item = /obj/item/storage/backpack/duffelbag/syndie/ammo/smg
 	cost = 20
 	include_modes = list(/datum/game_mode/nuclear)
+	
+/datum/uplink_item/ammo/carbine
+	name = "5.56 Toploader Magazine"
+	desc = "An additional 30-round 5.56 magazine; sutable for use with the M-90gl carbine. \
```
Comment (ShizCalev) [inline code/modules/uplink/uplink_items.dm]: 5.56mm

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by subject217** [comment]
For brevity's sake gonna point out that both of those descriptions were from the old version, but I'll change it.

File: code/modules/uplink/uplink_items.dm [hunk 3]
```diff
@@ -201,6 +201,15 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	cost = 10
 	surplus = 40
 	include_modes = list(/datum/game_mode/nuclear)
+	
+/datum/uplink_item/dangerous/carbine
+	name = "M-90gl Carbine"
+	desc = "A fully-loaded, specialized three-round burst carbine that fires 30-round 5.56mm magazines with a togglable \
+			underslung 40mm grenade launcher."
```
Comment (ShizCalev) [inline code/modules/uplink/uplink_items.dm]: `40mm underslung grenade launcher`

alternatively, `under-barrel`.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/uplink/uplink_items.dm [hunk 4]
```diff
@@ -508,6 +517,22 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	item = /obj/item/storage/backpack/duffelbag/syndie/ammo/smg
 	cost = 20
 	include_modes = list(/datum/game_mode/nuclear)
+	
+/datum/uplink_item/ammo/carbine
+	name = "5.56 Toploader Magazine"
```
Comment (ShizCalev) [inline code/modules/uplink/uplink_items.dm]: `5.56mm Top Loading Magazine`
A firearm loaded from the top is a top loader. The magazine itself would be top loading as it's feeding from the top of the weapon.

**Pull request merged**